app/src/main/java/com/example/next_show/fragments/FeedFragment.java

```diff
@@ -136,9 +147,41 @@ public boolean onNavigationItemSelected(@NonNull MenuItem item) {
                 }
             });
         }
+
+        // set up filtering by genre
+        setFilterButtons();
+
         return currView;
     }
 
+    private void setFilterButtons() {
+        // find the radio button group for filtering
+        rgGenre = currView.findViewById(R.id.rgGenre);
+
+        rgGenre.setOnCheckedChangeListener(new RadioGroup.OnCheckedChangeListener() {
+            @Override
+            public void onCheckedChanged(RadioGroup group, int checkedId) {
+                selectedButton = currView.findViewById(checkedId);
+                String buttonText = selectedButton.getText().toString().toLowerCase();
+
+                // check if radio button works
+                Log.i(TAG, "Selected: " + buttonText);
+
+                switch (buttonText) {
+                    case ACTION:
+                        adapter.filter(ACTION);
+                        break;
+                    case COMEDY:
+                        adapter.filter(COMEDY);
+                        break;
+                    case DRAMA:
+                        adapter.filter(DRAMA);
+                        break;
+                }
```

Comment:
Can someone click again to uncheck a button?

Reply:
Nope, unfortunately. I actually used checkboxes here initially but I can't get the radio buttons to uncheck themselves

Reply:
Would rec googling to see if you can find a way to uncheck a radio button